File: javascript/clock/clock.js
function at(hour, minute) {
  minute = minute || 0
  while (minute < 0) {
    hour -= 1
    minute += 60
  }
  while (hour < 0) hour += 24

  hour = (hour + Math.floor(minute / 60) ) % 24
  minute = minute % 60

  return {
    hour: hour,
    minute: minute,
    toString: function() {
      return pad(hour) + ':' + pad(minute)
    },
    equals: function(clock) {
      return this.toString() === clock.toString()
    },
    plus: function(minute) {
      return at(this.hour, this.minute + minute)
    },
    minus: function(minute) {
      return this.plus(-minute)
    }
  }
}

function pad(n) {
  return ('00' + n).substr(-2, 2)
}

module.exports = {
  at: at
};

File: go/clock/clock.go
package clock

import "fmt"

const testVersion = 4

type Clock struct {
	hour   int
	minute int
}

func New(hour, minute int) Clock {
	if minute >= 60 {
		hour += minute / 60
		minute = minute % 60
	}
	if hour >= 24 {
		hour = hour % 24
	}
	for minute < 0 {
		hour -= 1
		minute = minute + 60
	}
	for hour < 0 {
		hour = hour + 24
	}
	return Clock{hour: hour, minute: minute}
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

func (c Clock) Add(minutes int) Clock {
	return New(c.hour, c.minute+minutes)
}
